### PR TITLE
Closes #2130 - Improve ExpandProperty Documentation

### LINKED ISSUE #2130: Select-Object -ExpandProperty documentation must be improved
It is almost impossible to derive the actual behavior of the `-ExpandProperty` parameter from the [existing documentation](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Utility/Select-Object.md), especially with respect to combining it with `-Property`.

Here's the **_actual_ behavior**, as far as I can tell (correct me if I'm wrong):

* The values of the (invariably single) property whose name you pass to `-ExpandProperty` _become the output objects, with their original type_, if they are scalars.

   * If the referenced property contains a _collection_, and that collection is not of type `[object[]]`, the input collection type is lost, but the elements of that collection are again output with their original type, as an `[object[]]` array.

* Any properties you pass to `-Property` are _added as  `NoteProperty` members_ to the output objects.

  * If the `-ExpandProperty` property is a _collection_, the note properties are added to _each element_ of that collection (and the collection is invariably converted to `[object[]]` on output, as noted).  

  * Often, the output formatting may not _show_ these note properties, which makes it _look_ like they're not there (ignored), but they are.

---

It's hard to glean the above from the current topic:

> indicates that an attempt should be made to expand that property

That doesn't tell us more than the _name_ of the parameter already tells us.

> For example, if the specified property is an array, each value of the array is included in the output. If the property contains an object, the properties of that object are displayed in the output.

This is slightly misleading in that it's not about _including_  elements or properties, it's about making the value of the targeted property the _output object_: as-is in the case of a scalar, and as a `[object[]]` array in the case of a collection (whose elements are again output as-is).

As for Example 2 (`Get-Process Explorer | Select-Object -Property ProcessName -ExpandProperty Modules | Format-List`):

> The command uses the Property parameter of the Select-Object cmdlet to select the process names. Because the Modules property contains a ModuleProcess object that has many properties, the command uses the ExpandProperty parameter to get the properties of the objects in the Modules property of each process. 

What this command actually does:

* output an array of `[System.Diagnostics.ProcessModule]` instances, as contained in the `.Modules` property of the `System.Diagnostics.Process` instances output by `Get-Process`.

* with each such instance being decorated with the input process' `.ProcessName` property value.

Version(s) of document impacted
------------------------------
- [x] Impacts 6 document
- [x] Impacts 5.1 document
- [x] Impacts 5.0 document
- [x] Impacts 4.0 document
- [x] Impacts 3.0 document

### PR DESCRIPTION
**Pull request by bobbytreed**
Version(s) of document impacted
------------------------------
- [ ] Impacts 6.1 document
- [X] Impacts 6.0 document
- [X] Impacts 5.1 document
- [X] Impacts 5.0 document
- [X] Impacts 4.0 document
- [X] Impacts 3.0 document

Reason(s) for not updating all version of documents
--------------------------------------------------
- [ ] The documented feature was introduced in version (list version here) of PowerShell
- [ ] This issue only shows up in version (list version(s) here) of the document
- [ ] This PR partially fixes the issue, and issue #<insert here> tracks the remaining work

### CODE REVIEW

**Comment by zjalexander** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/6/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/6/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C6%5C19%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201806191445552913-2599%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=DWnJ5UTOnaEvgTqtDVtn%2FtNe%2BUgkvXuP2AJzvvNjYJg%3D&st=2018-06-19T14%3A51%3A30Z&se=2018-07-20T14%3A56%3A30Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by zjalexander** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/6/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/6/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C6%5C19%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201806191535247225-2599%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=LH1Gv7XmXBNg8ZLvvrzCiYiw7UAA2ieWaHx0Du0b6hA%3D&st=2018-06-19T15%3A40%3A14Z&se=2018-07-20T15%3A45%3A14Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md [hunk 1]
```diff
@@ -7,48 +7,59 @@ online version:  http://go.microsoft.com/fwlink/?LinkID=113387
...
+
+### Example 2: Select objects by property and format the results
+
+```powershell
+Get-Process Explorer | Select-Object -Property ProcessName -ExpandProperty Modules | Format-List
 ```
-PS C:\> Get-Process Explorer | Select-Object -Property ProcessName -ExpandProperty Modules | Format-List
 
+```output
 ProcessName       : 00THotkey
```
Comment (sdwheeler) [inline reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md]: Output doesn't match the command. I thought we had fixed this.

File: reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md [hunk 2]
```diff
@@ -100,71 +121,136 @@ ati2evxx     Thursday
...
+### Example 6: Select newest and oldest events in the event log
+
+```powershell
+$a = Get-EventLog -LogName "Windows PowerShell"
 ```
-PS C:\> $a = Get-Eventlog -Log "Windows PowerShell"
-$a | select-object -index 0, ($a.count - 1)
+
+```output
+PS C:\> $A | Select-Object -Index 0, ($A.count - 1)
```
Comment (sdwheeler) [inline reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md]: This is not output. It is another command.

File: reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md [hunk 3]
```diff
@@ -100,71 +125,136 @@ ati2evxx     Thursday
...
+
+### Example 6: Select newest and oldest events in the event log
+
+```powershell
+$a = Get-EventLog -LogName "Windows PowerShell"
 ```
-PS C:\> $a = Get-Eventlog -Log "Windows PowerShell"
-$a | select-object -index 0, ($a.count - 1)
+
+```powershell
```
Comment (sdwheeler) [inline reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md]: Merge the 2 code blocks. Make the case match for $a.  Make this change in all versions.

**Comment by zjalexander** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/6/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/6/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C6%5C19%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201806191626413001-2599%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=ZPmHSslAviP1WmfsxL0WWIWfeqwqKjG7s3%2FACL4nm68%3D&st=2018-06-19T16%3A31%3A34Z&se=2018-07-20T16%3A36%3A34Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by zjalexander** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/3.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/4.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.0/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/5.1/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |
[reference/6/Microsoft.PowerShell.Utility/Select-Object.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbyreed-issue2130/reference/6/Microsoft.PowerShell.Utility/Select-Object.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C6%5C19%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201806191650361579-2599%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=T%2BkwWNyzE3L0kv%2BOERt0Ii5dAUHU5z%2BwlCYCvSQlDZI%3D&st=2018-06-19T16%3A56%3A08Z&se=2018-07-20T17%3A01%3A08Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**